apps/screenshot-diff/src/computeScreenshotDiff.ts

```diff
@@ -54,19 +59,59 @@ export const computeScreenshotDiff = async (
     }),
   ]);
 
+  await Promise.all(
+    [
+      { ...screenshotDiff.baseScreenshot, filePath: baseScreenshotPath },
+      { ...screenshotDiff.compareScreenshot, filePath: compareScreenshotPath },
+    ]
+      .filter(
+        (screenshot) =>
+          screenshot &&
+          screenshot.fileId &&
+          screenshot.file &&
+          (screenshot.file.width === undefined ||
+            screenshot.file.height === undefined)
+      )
+      .map(async (screenshot) => {
+        const { width, height } = await getImageSize(screenshot.filePath);
+        await File.query()
+          .findById(screenshot.fileId as string)
+          .patch({ width, height });
+      })
+  );
+
+  if (
+    !screenshotDiff.baseScreenshot ||
+    (screenshotDiff.compareScreenshot!.fileId !== null &&
+      screenshotDiff.baseScreenshot!.fileId ===
+        screenshotDiff.compareScreenshot!.fileId)
+  ) {
+    await ScreenshotDiff.query().findById(screenshotDiff.id).patch({
+      jobStatus: "complete",
+    });
+    return;
+  }
+
   const difference = await diffImages({
     actualFilename: compareScreenshotPath,
     expectedFilename: baseScreenshotPath,
     diffFilename: diffResultPath,
   });
 
-  let uploadResult = null;
+  let file = null;
   if (difference.score > 0) {
-    uploadResult = await upload({
+    const uploadResult = await upload({
       s3,
       Bucket: bucket,
       inputPath: diffResultPath,
     });
+    file = await File.query()
```

Comment:
We don't have any transaction, so if the code failed after adding the file, we will have twice the same file in database. Since file key is unique it will break and the diff will never be processed.

Reply:
We have to insert file + screenshot diff in a transaction.